server/services/chat/olly_chat_service.ts

```diff
@@ -21,22 +25,89 @@ const INTERACTION_ID_FIELD = 'parent_interaction_id';
 
 export class OllyChatService implements ChatService {
   static abortControllers: Map<string, AbortController> = new Map();
+  private static rootAgentId: string | undefined;
 
-  private async requestAgentRun(
-    rootAgentId: string,
-    payload: AgentRunPayload,
-    context: RequestHandlerContext
-  ) {
+  constructor(
+    private readonly context: RequestHandlerContext,
+    private readonly rootAgentName: string
+  ) { }
+
+  private async initRootAgent() {
+    OllyChatService.rootAgentId = await this.searchRootAgent(this.rootAgentName);
+  }
+
+  private async searchRootAgent(rootAgentName: string): Promise<string> {
+    try {
+      const requestParams = {
+        query: {
+          term: {
+            'name.keyword': rootAgentName,
+          },
+        },
+        sort: {
+          created_time: 'desc',
+        },
+      };
+
+      const opensearchClient = this.context.core.opensearch.client.asCurrentUser;
+      const response = await opensearchClient.transport.request({
+        method: 'GET',
+        path: `${ML_COMMONS_BASE_API}/agents/_search`,
+        body: requestParams,
+      });
+
+      if (!response || response.body.hits.total.value === 0) {
+        throw new Error('cannot find any root agent by name: ' + rootAgentName);
+      }
+      return response.body.hits.hits[0]._id;
+    } catch (error) {
+      let errorMessage = JSON.stringify(error.meta?.body);
+      if (!errorMessage) {
+        errorMessage = error.toString();
```

Comment:
Since error is any here. The `error` may doesn't have `toString` method. How about refactor the errorMessage like this:
```ts
 const errorMessage = JSON.stringify(error.meta?.body || error);
```

Reply:
Thanks, by my test, if `error` is a instance of `Error`, then JSON.stringify(error) returns `{}` which is not as expected, but I found that `JSON.stringify(error.meta?.body) || error` works well.